File: src/main/java/platform/ecommerce/dto/response/LoginResponse.java
package platform.ecommerce.dto.response;

import lombok.Builder;

/**
 * Login response DTO containing tokens and member info.
 */
@Builder
public record LoginResponse(
        TokenResponse tokens,
        MemberResponse member
) {
    public static LoginResponse of(TokenResponse tokens, MemberResponse member) {
        return LoginResponse.builder()
                .tokens(tokens)
                .member(member)
                .build();
    }
}

File: src/main/kotlin/platform/ecommerce/dto/response/LoginResponse.kt
package platform.ecommerce.dto.response

import java.util.*

data class LoginResponse(
    val id: UUID,
    val email: String,
    val role: String,
    val accessToken: String,
    val refreshToken: String,
    val tokenType: String,
    val expiresIn: Long
) {
}